Code-4-Community/speak-for-the-trees-frontend
# Landing page and components for other map pages with sidebars

## Why

[Monday.com Ticket](https://code4community-team.monday.com/boards/702429655/views/12631042/pulses/788451785)

What benefit does this bring to the end user? Or, what benefit does this bring to developers working in the codebase?

This is the landing page for SFFT, it is the first thing that the user will see when entering app if they are not logged in.

This PR also contains reusable components for the map page and the content of the sidebar to make future map pages quicker to implement.

## This PR

Describe the changes required and any implementation choices you made to give context to reviewers.

Made a number of new components:

Landing.tsx the container for the entire page.

MapPage.tsx ageneric map page with the map component as the content and a sidebar with content that is passed in as props.

MapView.tsx the map (right now just an empty placeholder).

MapSideBar.tsx the sidebar component with a title and description given as props and content as a child. There are map pages with different kinds of sidebars so this implementation allows for that flexibility.

LandingTreeStats.tsx the information in the body of the sidebar which is passed as a child of the MapPage.

![landing-tree-stats](https://user-images.githubusercontent.com/19194912/97811088-41abd200-1c46-11eb-92d2-5b5da961ccc8.PNG)

MapCard.tsx outlined cards that take in a header and a body, for expressing small pieces of info. Will most likely add an onClick to the div around the card when implementing other map pages.

## Screenshots

Provide screenshots of any new components, styling changes, or pages.

Colors of the titles and the footer are not correct because I want to be sure to use the antd Title and Paragraph components. We should make a global theme and then simply add the correct class to each of these components. 

Tried to follow absolute scale of Figma designs closely but wound up with some scaling that might not be ideal if we are attempting to get the entire sidebar on one page. Wanted to confirm if we want to scale down and how.

FullScreen:
![landing](https://user-images.githubusercontent.com/19194912/97811266-5c327b00-1c47-11eb-91ce-be39ffa31f88.PNG)

Scaled down to 80%:
![landing-scaled](https://user-images.githubusercontent.com/19194912/97811260-563c9a00-1c47-11eb-97b4-6cff8cfff14e.PNG)


## Verification Steps

What steps did you take to verify your changes work? These should be clear enough for someone to be able to clone the branch and follow the steps themselves.

Mostly formatting and components, not complicated functions or ajax calls so verification was visual. Ran linting and prettier and will configure to work with travis before changing from a draft PR.
